src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -122,6 +125,13 @@ def cli_cosmosdb_create(cmd, client,
     if enable_public_network is not None:
         public_network_access = 'Enabled' if enable_public_network else 'Disabled'
 
+    system_assigned_identity = None
+    if assign_identity is not None:
+        if assign_identity:
```

Comment:
if user doesnt provide any value in default_identitythen we can assign
system_assigned_identity = ManagedServiceIdentity(type='None')